### PR TITLE
Remove Helm 2 support, add integration tests

### PR DESCRIPTION
**Pull request by mwalser**
This is the start of a series of changes for an upcoming major release of the helm-maven-plugin.

As first PR, the support for Helm 2 is removed, unused dependencies are removed from the pom.xml and integration tests are added.

### CODE REVIEW

File: src/test/it/package-lint-template/verify.groovy [hunk 1]
```diff
@@ -0,0 +1,13 @@
+def logLines = new File(basedir, "build.log").readLines()
+
+// Package goal
+assert logLines.any { line -> line.contains("Successfully packaged chart and saved it to") }
+
+// Lint goal
+assert logLines.any { line -> line.contains("Chart.yaml: icon is recommended") }
+assert logLines.any { line -> line.contains("1 chart(s) linted, 0 chart(s) failed") }
```
Comment (hho) [inline src/test/it/package-lint-template/verify.groovy:8]:
```suggestion
// Package goal
assert logLines.any { it.contains("Successfully packaged chart and saved it to") }

// Lint goal
assert logLines.any { it.contains("Chart.yaml: icon is recommended") }
assert logLines.any { it.contains("1 chart(s) linted, 0 chart(s) failed") }
```

File: src/main/kotlin/com/deviceinsight/helm/ResolveHelmMojo.kt [hunk 2]
```diff
@@ -60,85 +57,77 @@ open class ResolveHelmMojo : AbstractHelmMojo() {
...
+		checkHelmVersion()
+		helm = resolveHelmBinary()
+	}
 
-		val platformIdentifier = PlatformDetector.detectHelmReleasePlatformIdentifier()
-		val helmArtifact: Artifact =
-			repositorySystem.createArtifactWithClassifier(helmGroupId, helmArtifactId, helmVersion, "binary",
-				platformIdentifier)
+	private fun checkHelmVersion() {
+		val (majorVersion, minorVersion) = helmVersion.split('.').map(String::toInt)
```
Comment (hho) [inline src/main/kotlin/com/deviceinsight/helm/ResolveHelmMojo.kt:68]: Maybe add a sanity check (i.e. check that the format actually matches `\d+(\.\d+){2}`)?

File: pom.xml [hunk 3]
```diff
@@ -12,28 +12,30 @@
...
 	<properties>
+		<!-- Compiler -->
 		<kotlin.version>1.8.10</kotlin.version>
 		<kotlin.compiler.jvmTarget>1.8</kotlin.compiler.jvmTarget>
 		<maven.compiler.source>1.8</maven.compiler.source>
 		<maven.compiler.target>1.8</maven.compiler.target>
 		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 
+		<!-- Maven -->
 		<maven.version>3.6.2</maven.version>
```
Comment (hho) [inline pom.xml:23]: Maybe we should update to 3.9.1? (3.9 includes the update to JDK 8 as a baseline)

File: pom.xml [hunk 4]
```diff
@@ -12,28 +12,30 @@
...
-
-		<dokka-maven-plugin.version>1.7.20</dokka-maven-plugin.version>
+		<!-- Plugins -->
 		<maven-compiler-plugin.version>3.11.0</maven-compiler-plugin.version>
 		<maven-source-plugin.version>3.2.1</maven-source-plugin.version>
 		<gitflow-maven-plugin.version>1.19.0</gitflow-maven-plugin.version>
 		<nexus-staging-maven-plugin.version>1.6.13</nexus-staging-maven-plugin.version>
 		<maven-gpg-plugin.version>3.0.1</maven-gpg-plugin.version>
+		<maven-invoker-plugin.version>3.3.0</maven-invoker-plugin.version>
+		<maven-invoker-plugin.groovy.version>4.0.6</maven-invoker-plugin.groovy.version>
```
Comment (hho) [inline pom.xml:38]: Let's not add dependencies already out of date…
```suggestion
		<maven-invoker-plugin.version>3.5.0</maven-invoker-plugin.version>
		<maven-invoker-plugin.groovy.version>4.0.10</maven-invoker-plugin.groovy.version>
```

File: pom.xml [hunk 5]
```diff
@@ -220,21 +190,37 @@
...
+				<configuration>
+					<projectsDirectory>src/test/it</projectsDirectory>
+					<showErrors>true</showErrors>
+					<cloneProjectsTo>${project.build.directory}/its</cloneProjectsTo>
+					<pomIncludes>
+						<pomInclude>*/pom.xml</pomInclude>
+					</pomIncludes>
+					<settingsFile>src/test/it/settings.xml</settingsFile>
+					<localRepositoryPath>${project.build.directory}/it-repo</localRepositoryPath>
+					<postBuildHookScript>verify.groovy</postBuildHookScript>
```
Comment (hho) [inline pom.xml:212]: Any chance to do the tests in Kotlin instead? (e.g. Kotlin Scripting)

Or is this some hardcoded thing in the invoker plugin?

**Review by hho — CHANGES_REQUESTED** [review]
Nothing major, only some cleanups

**Review by mwalser — COMMENTED** [review]
@hho Thanks for the review, I applied your suggestions and also added a default value for Helm version

File: pom.xml (hunk 3 above)
Comment (hho) [reply pom.xml:23]: Given that the warning also appears on use of the current plugin with the new Maven, I think we'll have to address it at some point.
But OK, it doesn't have to be in this PR.

**Review by hho — APPROVED** [review]

**Pull request merged**